src/compiler/utilities.ts

```diff
@@ -438,6 +525,46 @@ module ts {
         }
     }
 
+    export function forEachYieldExpression(body: Block, visitor: (expr: YieldExpression) => void): void {

```

Comment:
`visitor` returns void and so does `forEachYieldExpression`. If we're doing this, maybe we should instead also change `forEachReturnStatement`, but the reason I wrote that the way I did is to stay true to the behavior of the general family of functions.